css-view-transitions-2/Overview.bs

```diff
@@ -264,6 +281,34 @@ Note: this event is fired when [=reveal document|revealing a document=].
 The <dfn attribute for=PageRevealEvent>viewTransition</dfn> [=getter steps=] are to return the
 [=inbound cross-document view-transition=] for [=this's=] [=relevant global object's=] [=associated document=].
 
+## Additions to {{Document}} ## {#additions-to-document-api}
+
+	<xmp class=idl>
+		dictionary StartViewTransitionOptions {
+			UpdateCallback? update = null;
+			sequence<DOMString>? types = null;
+		};
+
+		partial interface Document {
+
+			ViewTransition startViewTransition(optional (UpdateCallback or StartViewTransitionOptions)? callbackOrOptions = null);
```

Comment:
Do we need this IDL to say "callback or options"? L1 declares the API which only takes a callback : https://drafts.csswg.org/css-view-transitions-1/#additions-to-document-api. So if in L2 the IDL adds a new API which is

```
startViewTransition(StartViewTransitionOptions options = null);
```

Would that be equivalent?

Reply:
You can't have overloads like this in WebIDL (or in javascript, for that matter). The new method shadows the previous one in a way, but we make sure it's compatible.
Adding `startViewTransition(StartViewTransitionOptions options = null)` would shadow the L1 function and not allow using a callback argument.

Reply:
So, I was just playing around with the implementation and I think this is also not allowed due to the last point here:
https://webidl.spec.whatwg.org/#idl-nullable-type

> The [inner type](https://webidl.spec.whatwg.org/#dfn-inner-type) must not be:
> ...
> * a [union type](https://webidl.spec.whatwg.org/#dfn-union-type) that itself [includes a nullable type](https://webidl.spec.whatwg.org/#dfn-includes-a-nullable-type) or has a dictionary type as one of its [flattened member types](https://webidl.spec.whatwg.org/#dfn-flattened-union-member-types).

I don't quite understand the non-normative justification for this rule, but I also think that we can just make the union non-nullable?

Reply:
Specifically something like this might work
```
ViewTransition startViewTransition((UpdateCallback or StartViewTransitionOptions?) callbackOrOptionsOrNull);
```

Reply:
Fixed